### PR TITLE
Move header changes to dspace theme

### PR DESCRIPTION
**Pull request by artlowel**
## Description
While the changes to the header made in #1057 look good, they make the header a lot more difficult to theme. Putting everything on a single line works well with the square DSpace logo, however the vast majority of institutions have a rectangular logo. Simply changing the logo currently requires too much knowledge of angular and dspace to make it work.

This PR moves the header changes made in #1057 to the dspace theme, and puts the header of the base theme back the way it was before. I have kept the color changes in the base version, as they improve accessibility.

## Instructions for Reviewers
Test this PR by comparing the dspace theme with the current version deployed on demo7.dspace.org. It should look and work identical, on all screen sizes.

Then, enable the base theme, and verify that it looks and works properly on all screen sizes and that the logo is again on its own line. Try it out with a rectangular logo if you can.

---

**DSpace theme**  
Screen Shot 2021-06-01 at 13 38 33

---

**Base theme**  
Screen Shot 2021-06-01 at 13 39 19

---

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by tdonohue** [comment]
@artlowel : I took a second look at our odd CI problems, and it _looks to me_ like the version of ChromeDriver / Chrome changed right when this problem occurred.
* [Here's one of the last builds to succeed.](https://github.com/DSpace/dspace-angular/runs/2708960111?check_suite_focus=true)  It uses Google Chrome `v90.0.4430.212` (displayed in the "Install latest Chrome" step) and ChromeDriver `v90.0.4430.24` (displayed at the top of the "Run e2e tests" step).
* The build in this PR (and all recent PRs which are failing) are all now using Google Chrome `v91.0.4472.77` and ChromeDriver `v91.0.4472.19`.
* I attempted a rebuild of a PR today, and the failure is still occurring....so, that makes me wonder if v91 of ChromeDriver is somehow not working well with Jasmine.

I admit, I haven't had a chance yet to see if we can revert to an older version of Chrome/ChromeDriver to see if that fixes the issue.  I may be able to find time tomorrow though, if you don't get to it first.

**Comment by tdonohue** [comment]
Yes, after more digging, this seems to be a bug in the new ChromeDriver.  

It was reported to `angular/webdriver-manager` in https://github.com/angular/webdriver-manager/issues/497 and https://github.com/angular/webdriver-manager/issues/496

And here's the bug report in Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1205107

It sounds like work is in progress, but no fix has been released. The only workaround mentioned is to downgrade to Chrome/ChromeDriver v90 until this gets resolved.  I'll look into whether that's simple enough to do (temporarily).

**Comment by artlowel** [comment]
Thanks for sleuthing @tdonohue !

I tried disabling the step in build.yml that upgrades chrome, but v91 seems to be already part of the default image

**Comment by tdonohue** [comment]
@artlowel : I'm working on a possible fix to our GitHub Actions CI to pin both Chrome & ChromeDriver to v90 temporarily, since that seems like our best option for now.

**Comment by tdonohue** [comment]
@artlowel : If you could rebase this on the latest `main`, the CI/build fix has just been merged.  So it should resolve your build issues here.

**Comment by artlowel** [comment]
@tdonohue done

**Review by tdonohue — CHANGES_REQUESTED** [review]
@artlowel : Tested this today, and overall it looks good to me.  Just some minor inline things to clean up, as you accidentally removed some accessibility attributes from tags in the 'base' theme.  I think we should strive to keep accessibility level changes in the 'base' theme as much as possible.  Beyond that, this looks fine to me.

File: src/app/header/header.component.html [hunk 1]
```diff
@@ -1,24 +1,23 @@
-<header class="header">
-  <nav role="navigation" [attr.aria-label]="'nav.user.description' |translate" class="container navbar navbar-expand-md px-0">
-    <div class="d-flex flex-grow-1">
-      <a class="navbar-brand m-2" routerLink="/home">
-        <img src="assets/images/dspace-logo.svg" alt="logo"/>
+<header>
+  <div class="container">
+    <div class="d-flex flex-row justify-content-between">
+      <a class="navbar-brand my-2" routerLink="/home">
+        <img src="assets/images/dspace-logo.svg"/>
```
[the remark this answers is not in the document]
Comment (artlowel) [reply src/app/header/header.component.html]: Good catch. I just put the original version of the header back. I didn't think to add the same a11y improvements the new one had. Fixed now.

File: src/app/header/header.component.html [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-                aria-controls="collapsingNav"
-                aria-expanded="false" aria-label="Toggle navigation">
-          <span class="navbar-toggler-icon fas fa-bars fa-fw" aria-hidden="true"></span>
-        </button>
-      </div>
-    </div>
-  </nav>
-  <ds-themed-navbar></ds-themed-navbar>
 
+      <nav class="navbar navbar-light navbar-expand-md flex-shrink-0 px-0">
```
[the remark this answers is not in the document]
Comment (artlowel) [reply src/app/header/header.component.html]: Fixed

File: src/app/navbar/navbar.component.html [hunk 3]
```diff
@@ -1,24 +1,17 @@
...
-    </a>
-
-    <div id="collapsingNav" class="w-100 h-100">
-      <ul class="navbar-nav me-auto mb-2 mb-lg-0 h-100">
-        <ng-container *ngFor="let section of (sections | async)">
-          <ng-container
-            *ngComponentOutlet="(sectionMap$ | async).get(section.id).component; injector: (sectionMap$ | async).get(section.id).injector;"></ng-container>
-        </ng-container>
-      </ul>
+     class="navbar navbar-light navbar-expand-md p-md-0 navbar-container">    <!-- TODO remove navbar-container class when https://github.com/twbs/bootstrap/issues/24726 is fixed -->
```
[the remark this answers is not in the document]
Comment (artlowel) [reply src/app/navbar/navbar.component.html]: Fixed

**Review by tdonohue — APPROVED** [review]
👍 Thanks @artlowel ! This looks good to me now...and thanks for making the alt & aria attributes all i18n keys.

**Comment by tdonohue** [comment]
Merging this with `1 Approval` as discussed in last week's meeting.  If we notice any later issues they can be addressed in a follow-up ticket/PR.

**Pull request merged**